backend/substrapp/tasks/tasks.py

```diff
@@ -670,12 +674,17 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
                 logger.error(f'failed to fetch namespaced secrets {secret_namespace} with selector {label_selector}')
                 raise e
 
-            secrets = {s['metadata']['name']: int.from_bytes(b64decode(s['data']['key']), 'big')
-                       for s in secrets.to_dict()['items']}
+            secrets = {
+                s['metadata']['labels']['index']: list(b64decode(s['data']['key']))
+                for s in secrets.to_dict()['items']
+            }
```

Comment:
Can you explain the type change here  (from `int` to `list`) ?

Reply:
Answered on Slack.